Feat(Datamanagement): Remove most old case bundle files

patrikgrenfeldt opened the pull request:
## Description

### Added
- A command to remove all old Housekeeper bundle files that we have not decided to save

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [x] run `cg clean hk_case_bundle_files --dry-run`

### Expected test outcome
- [x] check that it lists files for the pipelines mip-dna, mip-rna, sarscov2 and microsalt
- [x] check that different versions of same bundle are processed
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [x] code approved by HS
- [x] tests executed by PG
- [x] "Merge and deploy" approved by PG
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- ~Document in Atlas: https://atlas.scilifelab.se/infrastructure/dataflow/data_management_plan/~ not needed until automated
- [ ] Deploy this branch on Hasta
- [ ] Inform to HS

henrikstranneheim commented on cg/cli/clean.py:
```diff
@@ -127,6 +188,103 @@ def scout_finished_cases(
+                version_file_tags: [str] = [tag.name for tag in version_file.tags]
+
+                has_protected_tags: bool = False
+                for protected_tags in protected_tags_lists:
```
I think this should be a function to improve readability and enable a unit test of the logic

patrikgrenfeldt replied:
refactored

henrikstranneheim commented on tests/cli/clean/test_hk_bundle_files.py:
```diff
@@ -0,0 +1,62 @@
+    timestamp: datetime,
+):
+    # GIVEN a housekeeper api with some alignment files
+    file_path = "path/to_file.cram"
+    tag = "cram"
+    hk_bundle_data = {
+        "name": case_id,
+        "created": timestamp,
+        "expires": timestamp,
+        "files": [
+            {"path": file_path, "archive": False, "tags": [case_id, tag]},
+        ],
+    }
+    store = cg_context.status_db
+    case = helpers.ensure_case(store=store, case_id=case_id)
+    helpers.add_analysis(store=store, case=case, started_at=timestamp, completed_at=timestamp)
+    helpers.ensure_hk_bundle(cg_context.housekeeper_api, bundle_data=hk_bundle_data)
```
Are there not already test fixtures for this that one can reuse?

patrikgrenfeldt replied:
fixture reused

henrikstranneheim commented on cg/cli/clean.py:
```diff
@@ -127,6 +188,103 @@ def scout_finished_cases(
+    """Clean up all files for all pipelines but those whitelisted"""
+    status_db: Store = context.obj.status_db
+    housekeeper_api: HousekeeperAPI = context.obj.housekeeper_api
+    before: datetime = datetime.now() - timedelta(days=days_old)
```
Should we turn this into a function that can be reused. Looks like you are testing the logic in a function (get_date_days_ago) only defined in the test? But not using an actual function here? I might have misunderstood something....

patrikgrenfeldt replied:
replaced with calls to the function

henrikstranneheim commented on tests/cli/clean/test_hk_case_bundle_files.py:
```diff
@@ -0,0 +1,163 @@
+    assert result.date() == yesterday.date()
+
+
+def get_date_days_ago(days_ago: int) -> dt.datetime:
+    return dt.datetime.now() - dt.timedelta(days=days_ago)
```
Why is this here and not imported from code that will be used when actually executing or added as a test fixture for reuse by other tests?

patrikgrenfeldt replied:
Moved

henrikstranneheim commented on tests/cli/clean/test_hk_case_bundle_files.py:
```diff
@@ -0,0 +1,163 @@
+    bundle_name: str = analysis.family.internal_id
+
+    # GIVEN a housekeeper api with some alignment files
+    file_path = "path/to_file.cram"
+    hk_bundle_data = {
+        "name": bundle_name,
+        "created": timestamp,
+        "expires": timestamp,
+        "files": [
+            {"path": file_path, "archive": False, "tags": [bundle_name, "cram"]},
+        ],
+    }
+    helpers.ensure_hk_bundle(cg_context.housekeeper_api, bundle_data=hk_bundle_data)
+
```
I think this would be nice to have as a fixture if it is not already

henrikstranneheim replied:
Any chance of making this as a fixture or reuse one already existing?

patrikgrenfeldt replied:
replaced with fixture

henrikstranneheim commented on tests/cli/clean/test_hk_case_bundle_files.py:
```diff
@@ -0,0 +1,163 @@
+
+    # GIVEN a housekeeper api with some alignment files with protected tags
+    protected_tags = PIPELINE_PROTECTED_TAGS[pipeline][0]
+    file_path = "path/to_file.fastq"
+    hk_bundle_data = {
+        "name": bundle_name,
+        "created": timestamp,
+        "expires": timestamp,
+        "files": [
+            {"path": file_path, "archive": False, "tags": protected_tags},
+        ],
+    }
+    helpers.ensure_hk_bundle(cg_context.housekeeper_api, bundle_data=hk_bundle_data)
```
yes definitively a fixture

henrikstranneheim reviewed (commented):
Nice!

patrikgrenfeldt commented on cg/cli/clean.py:
```diff
@@ -127,6 +188,103 @@ def scout_finished_cases(
+    before: datetime = datetime.now() - timedelta(days=days_old)
+
+    size_cleaned = 0
+    pipeline: Pipeline
```
As far as I know it's the only way to typehint the item in `for item in items:`

patrikgrenfeldt commented on cg/cli/clean.py:
```diff
@@ -27,7 +23,72 @@
+from housekeeper.store import models as hk_models
+from tabulate import tabulate
+
+PIPELINE_PROTECTED_TAGS = {
```
moved

patrikgrenfeldt commented on cg/cli/clean.py:
```diff
@@ -127,6 +188,103 @@ def scout_finished_cases(
+    default=365,
+    help="Clean all files with analysis dates older then given number of days",
+)
+@click.option("-d", "--dry-run", is_flag=True, help="Shows cases and files that would be cleaned")
```
Used now

patrikgrenfeldt commented on cg/cli/clean.py:
```diff
@@ -127,6 +188,103 @@ def scout_finished_cases(
+    pipeline: Pipeline
+    for pipeline in Pipeline:
+
+        protected_tags_lists = PIPELINE_PROTECTED_TAGS.get(pipeline)
```
named it workflow

patrikgrenfeldt commented on cg/cli/clean.py:
```diff
@@ -127,6 +188,103 @@ def scout_finished_cases(
+            ).all()
+
+            version_file: hk_models.File
+            for version_file in version_files:
```
added

henrikstranneheim commented on cg/apps/housekeeper/hk.py:
```diff
@@ -231,16 +233,21 @@ def new_tag(self, name: str, category: str = None):
         return self._store.tag(name)
 
+    @staticmethod
+    def get_tag_names_from_file(file: models.File) -> [str]:
```
```suggestion
    def get_tag_names_from_file(file: models.File) -> List[str]:
```

henrikstranneheim commented on cg/apps/housekeeper/hk.py:
```diff
@@ -231,16 +233,21 @@ def new_tag(self, name: str, category: str = None):
 
+    @staticmethod
+    def get_tag_names_from_file(file: models.File) -> [str]:
+        """Fetch a tag"""
```
```suggestion
        """Fetch tag names for a file"""
```

henrikstranneheim commented on cg/apps/housekeeper/hk.py:
```diff
@@ -254,10 +261,6 @@ def commit(self):
         """Wrap method in Housekeeper Store"""
         return self._store.commit()
 
-    def session_no_autoflush(self):
```
Not needed anymore! Or was it unused?

patrikgrenfeldt replied:
duplicated

henrikstranneheim commented on cg/cli/clean.py:
```diff
@@ -127,12 +133,47 @@ def scout_finished_cases(
+    ):
+
+        file_path: Path = Path(version_file.full_path)
+        file_size = file_path.stat().st_size
```
```suggestion
        file_size: int = file_path.stat().st_size
```

henrikstranneheim commented on cg/meta/clean/api.py:
```diff
@@ -0,0 +1,111 @@
+        """Check if a file has any protected tags"""
+
+        LOG.info(f"File {file.full_path} has the tags {file.tags}")
+        version_file_tags: List[str] = HousekeeperAPI.get_tag_names_from_file(file)
```
```suggestion
        version_file_tag_names: List[str] = HousekeeperAPI.get_tag_names_from_file(file)
```
Why do we have the "version_" in the variable name here?

patrikgrenfeldt replied:
it doesn't make sense in this context anymore, renamed now

henrikstranneheim commented on cg/meta/clean/api.py:
```diff
@@ -0,0 +1,111 @@
+        for protected_tags in protected_tags_lists:
+
+            protected_tags_on_file = set(version_file_tags).intersection(protected_tags)
+            if protected_tags_on_file:
+                if protected_tags_on_file == set(protected_tags):
```
```suggestion
            if protected_tags_on_file and protected_tags_on_file == set(protected_tags):
```
Less deep - what do you think

patrikgrenfeldt replied:
refactored

henrikstranneheim commented on cg/meta/clean/api.py:
```diff
@@ -0,0 +1,111 @@
+        return _has_protected_tags
+
+    def get_unprotected_existing_bundle_files(self, before: datetime) -> Iterator[hk_models.File]:
+        """Returns all existing bundle files from analyses started before 'before' that hos no protected tags"""
```
```suggestion
        """Returns all existing bundle files from analyses started before 'before' that have no protected tags"""
```

henrikstranneheim commented on cg/meta/clean/api.py:
```diff
@@ -0,0 +1,111 @@
+            analysis: models.Analysis
+            for analysis in analyses:
+                LOG.info("Cleaning analysis %s", analysis)
+                version_files: List[hk_models.File] = self.get_bundle_files(
```
```suggestion
                hk_version_files: List[hk_models.File] = self.get_bundle_files(
```
To specify a bit more what kind of version we are describing

patrikgrenfeldt replied:
renamed

henrikstranneheim commented on tests/apps/hk/test_file.py:
```diff
@@ -170,3 +170,19 @@ def test_check_bundle_files(
+    # WHEN fetching tags of a file
+    tag_names = populated_housekeeper_api.get_tag_names_from_file(file_obj)
+
+    # THEN is a list of tag names
```
```suggestion
    # THEN a list of tag names is returned
```

henrikstranneheim commented on tests/apps/hk/test_file.py:
```diff
@@ -170,3 +170,19 @@ def test_check_bundle_files(
+
+    # THEN is a list of tag names
+    assert tag_names is not None
+    # THEN type is list of strings
```
```suggestion
    # THEN the return type is a list of strings
```

henrikstranneheim reviewed (approved):
Nice to move the clean code to a separate API 👍

patrikgrenfeldt commented on cg/cli/clean.py:
```diff
@@ -127,6 +188,103 @@ def scout_finished_cases(
+                version_file_tags: [str] = [tag.name for tag in version_file.tags]
+
+                has_protected_tags: bool = False
+                for protected_tags in protected_tags_lists:
```
I think it might confuse to have different wording on left and right side in a `for item in items` so I kind of prefer `for protected_tags in protected_tags_lists` to `for file_protected_tags in pipeline_protected_tags`

henrikstranneheim reviewed (approved):
👍

patrikgrenfeldt commented:
Test result:
```
/home/proj/bin/conda/envs/S_main/bin/cg --config /home/proj/production/servers/config/hasta.scilifelab.se/cg.yaml clean hk-case-bundle-files --dry-run --days-old 1995
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Fetch version 2016-09-07 13:54:05 from bundle epicghost
Version found for bundle:epicghost; pipeline: mip-dna; date 2016-09-07 13:54:05
Fetching files from bundle epicghost
Fetching files from version 172
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/query.py:2577: SAWarning: Pathed join target File.version has already been joined to; skipping
  "been joined to; skipping" % prop
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_config.yaml has the tags [<Tag(id=2, name='config', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_qc_sampleInfo.yaml has the tags [<Tag(id=3, name='sampleinfo', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt has the tags [<Tag(id=4, name='pedigree', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt has no protected tags.
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt found on disk.
Dry run: True. Keeping file /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_BOTH.bcf has the tags [<Tag(id=5, name='bcf-raw', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_BOTH.bcf.csi has the tags [<Tag(id=6, name='bcf-raw-index', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_vt_vep_parsed_snpeff_ranked_BOTH.selected.vcf has the tags [<Tag(id=7, name='vcf-clinical', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_vt_vep_parsed_snpeff_ranked_BOTH.vcf has the tags [<Tag(id=8, name='vcf-research', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/mip.pl_2016-09-07T13:54:05.log has the tags [<Tag(id=9, name='log', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml has the tags [<Tag(id=10, name='meta', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml has no protected tags.
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml found on disk.
Dry run: True. Keeping file /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_SV.bcf has the tags [<Tag(id=11, name='bcf-raw-sv', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_vep_parsed_ranked_SV.selected.vcf has the tags [<Tag(id=12, name='vcf-clinical-sv', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_vep_parsed_ranked_SV.vcf has the tags [<Tag(id=13, name='vcf-research-sv', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_qcmetrics.mod.yaml has the tags [<Tag(id=14, name='qc', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
Process freed 0.0 GB. Dry run: True
```
🆗 only meta and pedigree marked for deletion

patrikgrenfeldt commented:
Test result deletion:
```
/home/proj/bin/conda/envs/S_main/bin/cg --config /home/proj/production/servers/config/hasta.scilifelab.se/cg.yaml clean hk-case-bundle-files --days-old 1995
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Fetch version 2016-09-07 13:54:05 from bundle epicghost
Version found for bundle:epicghost; pipeline: mip-dna; date 2016-09-07 13:54:05
Fetching files from bundle epicghost
Fetching files from version 172
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/query.py:2577: SAWarning: Pathed join target File.version has already been joined to; skipping
  "been joined to; skipping" % prop
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_config.yaml has the tags [<Tag(id=2, name='config', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_qc_sampleInfo.yaml has the tags [<Tag(id=3, name='sampleinfo', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt has the tags [<Tag(id=4, name='pedigree', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt has no protected tags.
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt found on disk.
Fetching file 1509
Deleting file 1509 from housekeeper
Removed file /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt. Dry run: False
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_BOTH.bcf has the tags [<Tag(id=5, name='bcf-raw', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_BOTH.bcf.csi has the tags [<Tag(id=6, name='bcf-raw-index', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_vt_vep_parsed_snpeff_ranked_BOTH.selected.vcf has the tags [<Tag(id=7, name='vcf-clinical', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_vt_vep_parsed_snpeff_ranked_BOTH.vcf has the tags [<Tag(id=8, name='vcf-research', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/mip.pl_2016-09-07T13:54:05.log has the tags [<Tag(id=9, name='log', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml has the tags [<Tag(id=10, name='meta', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml has no protected tags.
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml found on disk.
Fetching file 1515
Deleting file 1515 from housekeeper
Removed file /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml. Dry run: False
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_SV.bcf has the tags [<Tag(id=11, name='bcf-raw-sv', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_vep_parsed_ranked_SV.selected.vcf has the tags [<Tag(id=12, name='vcf-clinical-sv', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_vep_parsed_ranked_SV.vcf has the tags [<Tag(id=13, name='vcf-research-sv', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
File /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_qcmetrics.mod.yaml has the tags [<Tag(id=14, name='qc', category=None, created_at=datetime.datetime(2017, 10, 2, 14, 16, 27))>]
Process freed 0.0 GB. Dry run: False
```
```
housekeeper get bundle epicghost -v
┏━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
┃ ID   ┃ File name                                                                                                                                                  ┃ Tags            ┃
┡━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
│ 1507 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_config.yaml                                                                           │ config          │
│ 1508 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_qc_sampleInfo.yaml                                                                    │ sampleinfo      │
│ 1510 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_BOTH.bcf                                      │ bcf-raw         │
│ 1511 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_BOTH.bcf.csi                                  │ bcf-raw-index   │
│ 1512 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_vt_vep_parsed_snpeff_ranked_BOTH.selected.vcf │ vcf-clinical    │
│ 1513 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_gvcf_vrecal_comb_vt_vep_parsed_snpeff_ranked_BOTH.vcf          │ vcf-research    │
│ 1514 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/mip.pl_2016-09-07T13:54:05.log                                                              │ log             │
│ 1516 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_SV.bcf                                                    │ bcf-raw-sv      │
│ 1517 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_vep_parsed_ranked_SV.selected.vcf                         │ vcf-clinical-sv │
│ 1518 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_sorted_md_rreal_brecal_comb_vep_parsed_ranked_SV.vcf                                  │ vcf-research-sv │
│ 1519 │ /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_qcmetrics.mod.yaml                                                                    │ qc              │
└──────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┴─────────────────┘
[0|0|0] [patrik.grenfeldt@hasta:/home/proj/production/servers/resources/hasta.scilifelab.se] [P_main] master 7s ± ll /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml
ls: cannot access /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/meta.yaml: No such file or directory
[0|0|0] [patrik.grenfeldt@hasta:/home/proj/production/servers/resources/hasta.scilifelab.se] [P_main] master 2 ± ll /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt
ls: cannot access /home/proj/production/housekeeper-bundles/epicghost/2016-09-07/16108_pedigree.txt: No such file or directory
```
🆗 files deleted on disk and in housekeeper

patrikgrenfeldt commented:
Test result different pipelines:
```
[0|0|0] [patrik.grenfeldt@hasta:/home … ] [S_main] master ± cg clean hk-case-bundle-files --dry-run --days-old ...> ~/test-clean.txt 2>&1
[S_main] master ± grep mip-dna ~/test-clean.txt | wc -l
1343
[0|0|0] [patrik.grenfeldt@hasta:/home … ] [S_main] master ± grep mip-rna ~/test-clean.txt | wc -l
165
[0|0|0] [patrik.grenfeldt@hasta:/home … ] [S_main] master ± grep microsalt ~/test-clean.txt | wc -l
807
[0|0|1787] θ60° [patrik.grenfeldt@hasta:/home … ] [S_main] master ± grep cov-2 ~/test-clean.txt | wc -l
150
```
🆗 All pipelines with tags defined as protected in result

patrikgrenfeldt commented:
Deployed to Hasta:
<img width="320" alt="image" src="https://user-images.githubusercontent.com/6697636/157235804-ab69750f-6fb4-4a23-9a04-da066713686a.png">

patrikgrenfeldt commented:
@henrikstranneheim [Feat(Datamanagement): Remove most old case bundle files](https://github.com/Clinical-Genomics/cg/pull/1418#) has been deployed!

henrikstranneheim commented:
@patrikgrenfeldt Nice work!
